src/App.js

```diff
@@ -1,25 +1,159 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import "./SlotMachine.css";
+
+const symbols = ["🍍", "🐠", "🥥", "🌴", "🐒", "🏴‍☠️"];
+const credits = [1, 5, 10, 20, 50, 100];
+
+function getRandomSymbol() {
+  return symbols[Math.floor(Math.random() * symbols.length)];
+}
+
+const SlotMachine = () => {
+  const [reels, setReels] = useState(
+    Array(3)
+      .fill(null)
+      .map(() => Array(3).fill(symbols[2]))
+  );
+  const [isSpinning, setIsSpinning] = useState(false);
+  const [message, setMessage] = useState("Deposit to play!");
+  const [balance, setBalance] = useState(100);
+  const [bet, setBet] = useState(credits[0]);
+  const [spins, setSpins] = useState(0);
+  const [wins, setWins] = useState(0);
+  const [losses, setLosses] = useState(0);
+
+  // Spins the reels
+  const spinReels = () => {
+    if (bet > balance) {
+      setMessage("Insufficient balance to spin!");
+      return;
+    }
+
+    setIsSpinning(true);
```

Comment:
keep in mind that every time setState sets an updated state, the component rerenders. This is 4 rerenders in one go. Something to note and be mindful of. The app is so small and the rerenders are fast so you won't be affected, but if we're thinking of scaling this app and adding more features, this can start to become inefficient.